### ISSUE TITLE
JAVA_HOME vs java.home

### ISSUE THREAD

**Issue by Andrei-Pozolotin** [davidB/scala-maven-plugin#221]
combination of scala-maven-plugin v 3.3.1 and maven v 3.5.0
produces error (did not occur few plugin/maven versions back)
```
[INFO] Using zinc server for incremental compilation
[INFO] Toolchain in scala-maven-plugin: /usr/lib/jvm/java-8-oracle/jre
[warn] Pruning sources from previous analysis, due to incompatible CompileSetup.
[info] Compiling 9 Scala sources and 1 Java source to /home/work/source/git/carrot-project/base/conf/target/classes...
[error] Cannot run program "/usr/lib/jvm/java-8-oracle/jre/bin/javac": error=2, No such file or directory
```
basically:
* maven treats `java.home` as JRE
* but plugin expects there to find JDK, see recent commits `JavaMainCallerByFork`

```
echo $JAVA_HOME
/usr/lib/jvm/java-8-oracle
```
```
mvn -v
Apache Maven 3.5.0 (NON-CANONICAL_2017-04-10T13:56:20+03:00_root; 2017-04-10T05:56:20-05:00)
Maven home: /opt/maven
Java version: 1.8.0_152, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-8-oracle/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.13.8-1-arch", arch: "amd64", family: "unix"
```

**Comment by Andrei-Pozolotin** [comment]
possible change in `mvn` code for `JAVA_HOME`:
https://maven.apache.org/docs/3.5.0/release-notes.html

**Comment by Andrei-Pozolotin** [comment]
the work around is to force use toolchains (again, see `JavaMainCallerByFork`):

https://maven.apache.org/guides/mini/guide-using-toolchains.html
```
# $HOME/.m2/toolchains.xml
<?xml version="1.0" encoding="UTF8"?>
<toolchains>
  <!-- JDK toolchains -->
  <toolchain>
    <type>jdk</type>
    <provides>
      <version>1.8</version>
      <vendor>sun</vendor>
    </provides>
    <configuration>
      <jdkHome>/usr/lib/jvm/default</jdkHome>
    </configuration>
  </toolchain>
</toolchains>
```

```
# project/pom.xml
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-toolchains-plugin</artifactId>
				<version>1.1</version>
				<executions>
					<execution>
						<goals>
							<goal>toolchain</goal>
						</goals>
					</execution>
				</executions>
				<configuration>
					<toolchains>
						<jdk>
							<version>[1.8,)</version>
						</jdk>
					</toolchains>
				</configuration>
			</plugin>
```

**Comment by nikitaporubovispb** [comment]
Same bug with version 4.4.0 to 4.5.4
```
p@p:~/git/P$ mvn -v
Apache Maven 3.6.3
Maven home: /usr/share/maven
Java version: 1.8.0_212, vendor: Oracle Corporation, runtime: /usr/lib/jvm/jdk1.8.0_212/jre
Default locale: ru_RU, platform encoding: UTF-8
OS name: "linux", version: "5.4.0-86-generic", arch: "amd64", family: "unix"
```
```
p@p:~/git/P$ echo $JAVA_HOME
/usr/lib/jvm/jdk1.8.0_212
```
```
  Выбор   Путь                                     Приор Состояние
------------------------------------------------------------
  0            /usr/lib/jvm/java-11-openjdk-amd64/bin/java   1111      автоматический режим
  1            /usr/lib/jvm/java-11-openjdk-amd64/bin/java   1111      ручной режим
  2            /usr/lib/jvm/jdk1.6.0_45/bin/java             1065      ручной режим
  3            /usr/lib/jvm/jdk1.7.0_80/bin/java             1         ручной режим
* 4            /usr/lib/jvm/jdk1.8.0_212/bin/java            2         ручной режим
```
```
[ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:4.5.4:compile (scala-compile-first) on project p4.model: wrap: java.io.IOException: Cannot run program "/usr/lib/jvm/jdk1.8.0_212/jre/bin/javac" (in directory "/home/p/git/P"): error=2, Нет такого файла или каталога -> [Help 1]
```
```
[ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:4.4.0:compile (scala-compile-first) on project p4.model: wrap: java.io.IOException: Cannot run program "/usr/lib/jvm/jdk1.8.0_212/jre/bin/javac" (in directory "/home/p/git/P"): error=2, Нет такого файла или каталога -> [Help 1]
```

**Comment by chocolate213** [comment]
> Same bug with version 4.4.0 to 4.5.4
> 
> ```
> p@p:~/git/P$ mvn -v
> Apache Maven 3.6.3
> Maven home: /usr/share/maven
> Java version: 1.8.0_212, vendor: Oracle Corporation, runtime: /usr/lib/jvm/jdk1.8.0_212/jre
> Default locale: ru_RU, platform encoding: UTF-8
> OS name: "linux", version: "5.4.0-86-generic", arch: "amd64", family: "unix"
> ```
> 
> ```
> p@p:~/git/P$ echo $JAVA_HOME
> /usr/lib/jvm/jdk1.8.0_212
> ```
> 
> ```
>   Выбор   Путь                                     Приор Состояние
> ------------------------------------------------------------
>   0            /usr/lib/jvm/java-11-openjdk-amd64/bin/java   1111      автоматический режим
>   1            /usr/lib/jvm/java-11-openjdk-amd64/bin/java   1111      ручной режим
>   2            /usr/lib/jvm/jdk1.6.0_45/bin/java             1065      ручной режим
>   3            /usr/lib/jvm/jdk1.7.0_80/bin/java             1         ручной режим
> * 4            /usr/lib/jvm/jdk1.8.0_212/bin/java            2         ручной режим
> ```
> 
> ```
> [ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:4.5.4:compile (scala-compile-first) on project p4.model: wrap: java.io.IOException: Cannot run program "/usr/lib/jvm/jdk1.8.0_212/jre/bin/javac" (in directory "/home/p/git/P"): error=2, Нет такого файла или каталога -> [Help 1]
> ```
> 
> ```
> [ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:4.4.0:compile (scala-compile-first) on project p4.model: wrap: java.io.IOException: Cannot run program "/usr/lib/jvm/jdk1.8.0_212/jre/bin/javac" (in directory "/home/p/git/P"): error=2, Нет такого файла или каталога -> [Help 1]
> ```

Fix at 4.7.0
